File: server/php/src/Application/Domain/Payments/PaymentDetails/PaymentProduct840CustomerAccount.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class PaymentProduct840CustomerAccount
{
    public ?string $accountId = null;
    public ?string $companyName = null;
    public ?string $countryCode = null;
    public ?string $customerAccountStatus = null;
    public ?string $customerAddressStatus = null;
    public ?string $firstName = null;
    public ?string $payerId = null;
    public ?string $surname = null;

    public function toArray(): array
    {
        return [
            'accountId' => $this->accountId,
            'companyName' => $this->companyName,
            'countryCode' => $this->countryCode,
            'customerAccountStatus' => $this->customerAccountStatus,
            'customerAddressStatus' => $this->customerAddressStatus,
            'firstName' => $this->firstName,
            'payerId' => $this->payerId,
            'surname' => $this->surname,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/payment_product840_customer_account.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class PaymentProduct840CustomerAccount
          attr_accessor :account_id, :company_name, :country_code, :customer_account_status,
                        :customer_address_status, :first_name, :payer_id, :surname

          def initialize(account_id: nil, company_name: nil, country_code: nil,
                         customer_account_status: nil, customer_address_status: nil,
                         first_name: nil, payer_id: nil, surname: nil)
            @account_id = account_id
            @company_name = company_name
            @country_code = country_code
            @customer_account_status = customer_account_status
            @customer_address_status = customer_address_status
            @first_name = first_name
            @payer_id = payer_id
            @surname = surname
          end

          def to_h
            {
              'accountId' => @account_id,
              'companyName' => @company_name,
              'countryCode' => @country_code,
              'customerAccountStatus' => @customer_account_status,
              'customerAddressStatus' => @customer_address_status,
              'firstName' => @first_name,
              'payerId' => @payer_id,
              'surname' => @surname
            }
          end
        end
      end
    end
  end
end
